style: apply /style-guide pass to models/tables (#2725)
wandb/docs: models/tables/tables-walkthrough.mdx

---
description: Explore how to use W&B Tables with this 5-minute Quickstart.
title: 'Tutorial: Log tables, visualize, and query data'
keywords: ["wandb.Table", "Pandas DataFrame", "cross-run comparison", "table visualization"]
---

> This Quickstart walks you through logging data tables to W&B, visualizing them in your project workspace, and comparing results across runs. By the end, you have a logged table that you can explore and query in the W&B App.

Author: Addressed in 7910ed9 by removing the unfulfilled "query" promise from the intro ("explore and query" → "explore and compare"), which matches what the walkthrough actually demonstrates: log → visualize → cross-run compare.

I deliberately kept the page **title** ("...visualize, and query data") because it's referenced externally — `examples.mdx`, `models/ref/python/experiments/run.mdx`, and the localized mirrors all link to this page by that name — and querying a logged table is supported in the App and documented on [Visualize and analyze tables](/models/tables/visualize-tables). Happy to instead add a short query step so the page fully lives up to the title, if you'd prefer that over the framing fix.

---
description: Explore how to use W&B Tables with this 5-minute Quickstart.
title: 'Tutorial: Log tables, visualize, and query data'
---

import { ColabLink } from '/snippets/_includes/colab-link.mdx';

> This page walks you through logging data tables to W&B, visualizing them in your project workspace, and comparing results across runs. By the end, you have a logged table that you can explore and compare in W&B.

Reviewer: ```suggestion
Learn how to log data tables to W&B, visualize them in your project workspace, and compare results across runs. By the end, you’ll have a logged table that you can explore and compare in W&B.
```

> This Quickstart walks you through logging data tables to W&B, visualizing them in your project workspace, and comparing results across runs. By the end, you have a logged table that you can explore and compare in the W&B App.

Reviewer 2: ```suggestion
This page walks you through logging data tables to W&B, visualizing them in your project workspace, and comparing results across runs. By the end, you have a logged table that you can explore and compare in W&B.
```

Author: Applied: "This page walks you..." and "in W&B".

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

Select the button below to try a PyTorch Quickstart example project on MNIST data.

<ColabLink url="https://colab.research.google.com/github/wandb/examples/blob/master/colabs/datasets-predictions/W%26B_Tables_Quickstart.ipynb" />

## Log a table

> In this step, you create a table and log it to W&B so that it's available for visualization later in the walkthrough. You can either construct a new table or pass a Pandas DataFrame.

Reviewer: ```suggestion
In this step, you create a table and log it to W&B so that it's available for visualization later in the walkthrough. You can either construct a new table or pass a pandas DataFrame.
```

    run.log({"Table Name": my_table})
```
</Tab>
<Tab title="Pandas DataFrame">

> Pass a Pandas DataFrame to `wandb.Table()` to create a new table.

Reviewer: ```suggestion
Pass a pandas DataFrame to `wandb.Table()` to create a new table.
```

</Tabs>

## Visualize tables in your project workspace

> After logging a table, you can view it in W&B to confirm it was recorded correctly and to explore its contents.

Reviewer: This makes it sound like we'll fumble displaying the a user's table

```suggestion
After logging a table, you can view it in W&B.
```

1. Navigate to your project in W&B.
2. Select the name of your run in your project workspace. W&B adds a new panel for each unique table key.

<Frame>
    <img src="/images/data_vis/wandb_demo_logged_sample_table.png" alt="Sample table logged" />
</Frame>

In this example, `my_table` is logged under the key `"Table Name"`.

## Compare across model versions

After you have logged tables from more than one run, you can use the project workspace to compare results side by side and evaluate how model versions differ.

> Log sample tables from multiple W&B Runs and compare results in the project workspace. This [example workspace](https://wandb.ai/carey/table-test?workspace=user-carey) shows how to combine rows from multiple different versions in the same table.

Reviewer: ```suggestion
After you log tables from multiple W&B runs, compare results side by side in the project workspace to evaluate differences between model versions. This [example workspace](https://wandb.ai/carey/table-test?workspace=user-carey) shows how to combine rows from multiple versions in the same table.
```
